R/run-nmtran.R

```diff
@@ -0,0 +1,330 @@
+#' Interface for running `NM-TRAN` on model objects
+#'
+#' Function to run `NM-TRAN` on a model object to validate its control stream
+#'  for correct coding before submission. The `NM-TRAN` dataset (`FDATA`) and
+#'  other `NONMEM` artifacts can be further inspected by keeping the run directory
+#'  around.
+#'
+#' @details
+#' `NM-TRAN` is a preprocessor for `NONMEM` that translates user-specified
+#' control stream data and instructions into a form executable by `NONMEM`.
+#'
+#' Note that `nmtran_presort` is run ahead of `NM-TRAN` for `NONMEM` versions
+#' `'nm74gf'`, `'nm74gf_nmfe'`, and `'nm75'`.
+#'  - `nmtran_presort` is an supplementary utility that preprocesses the control
+#'     stream to ensure it is in the correct format for `NM-TRAN`. It is particularly
+#'     relevant for handling specific data manipulations and ensuring compatibility
+#'     with the `NM-TRAN` executable.
+#'
+#' @param .mod A `bbi_nonmem_model` object.
+#' @param .bbi_args A named list specifying arguments to pass to `NM-TRAN`.
+#'   Similar to the `.bbi_args` argument defined in [submit_model()], though here
+#'   only `prdefault`, `tprdefault`, and `maxlim` flags are passed to `NM-TRAN`.
+#'   See [print_bbi_args()] for more details.
+#' @inheritParams submit_model
+#' @param .nonmem_version Character scalar for default version of NONMEM to use.
+#'   If left `NULL`, will look for the default version specified in the provided
+#'   `bbi` configuration file.
+#' @param clean Logical (`T`/`F`). If `FALSE`, don't delete the temporary folder
+#'   containing the `NM-TRAN` run, which will be stored in the current working
+#'   directory.
+#' @param run_dir Directory to run `NM-TRAN` in. Only relevant if `clean = FALSE`.
+#'
+#' @examples
+#' \dontrun{
+#'
+#' mod <- read_model(file.path(MODEL_DIR, 1))
+#' run_nmtran(mod, .nonmem_version = "nm75")
+#'
+#' # Save the run directory for manual inspection
+#' run_nmtran(mod, clean = FALSE, run_dir = getwd())
+#'
+#' }
+#'
+#' @export
+run_nmtran <- function(
+    .mod,
+    .bbi_args = list(prdefault = FALSE, tprdefault = FALSE, maxlim = 2),
+    .nonmem_version = NULL,
```

Comment:
This goes hand in hand with the previous comment. To align with `submit_model`, I think `.nonmem_version` parameter should dropped and the value should be taken from the "nm\_version" specified in `.bbi_args`, if any.

In other words, I think there's value in being able to invoke `run_nmtran` with a subset of `submit_model` arguments.

```R
bbi_args <- list(...)
config < "..."

run_nmtran(mod, .bbi_args = bbi_args, .config_path = config)
submit_model(mod, .bbi_args = bbi_args, .config_path = config, ...)
```

Reply:
I handled this by making a minor adjustment to the `nmtran_setup` defaults and just not exposing this argument to the user via `run_nmtran`. Will leave this as unresolved for now in the event we want a different approach.